src/app/profile/settings/feature-opt-in/feature-opt-in.component.spec.ts

```diff
@@ -20,11 +22,39 @@ class HostComponent {}
 describe('FeatureOptInComponent', () => {
   type Context = TestContext<FeatureOptInComponent, HostComponent>;
 
+  let featureOptInComponent: FeatureOptInComponent;
+  let gettingStartedServiceMock: jasmine.SpyObj<GettingStartedService>;
+  let notificationsMock: jasmine.SpyObj<Notifications>;
+  let userServiceMock: jasmine.SpyObj<UserService>;
+  let toggleServiceMock: jasmine.SpyObj<FeatureTogglesService>;
+  let toggleAckServiceMock: jasmine.SpyObj<FeatureAcknowledgementService>;
+
+  beforeEach(() => {
+    gettingStartedServiceMock = createMock(GettingStartedService);
+    toggleServiceMock = createMock(FeatureTogglesService);
+    toggleAckServiceMock = createMock(FeatureAcknowledgementService);
+    notificationsMock = createMock(Notifications);
+    userServiceMock = createMock(UserService);
+    (userServiceMock as any).currentLoggedInUser = {
+      attributes: {
+        featureLevel: 'beta',
+        email: 'somebody@email.com',
+        emailVerified: true
+      }
+    };
+    toggleServiceMock.getAllFeaturesEnabledByLevel.and.returnValue(Observable.of([]));
+    toggleAckServiceMock.getToggle.and.returnValue(Observable.of(true));
+    gettingStartedServiceMock.createTransientProfile.and.returnValue({ featureLevel: 'beta' } as ExtProfile);
+    gettingStartedServiceMock.update.and.returnValue(Observable.of({}));
+    notificationsMock.message.and.returnValue(Observable.of({}));
```

Comment:
These mocks don't appear to be actually used. You've declared them above and instantiate them in this beforeEach, but they aren't provided in the module metadata object lower down. The module metadata has a bunch of "useFactory"s that already provide other instances of these mocks. The only one that's actually being added here is the toggleAckServiceMock. If the "and.returnValue" stubbing really needs to be added here, please do it by moving those stubbing lines into the provider factories below. Otherwise, please simply remove these additional unused declarations and definitions.